recipes/CommonVoice/ASR/seq2seq/train_with_wav2vec.py

```diff
@@ -63,30 +63,30 @@ def compute_forward(self, batch, stage):
         p_seq = self.hparams.log_softmax(logits)
 
         # Compute outputs
+        p_ctc, p_tokens = None, None
         if stage == sb.Stage.TRAIN:
             current_epoch = self.hparams.epoch_counter.current
             if current_epoch <= self.hparams.number_of_ctc_epochs:
                 # Output layer for ctc log-probabilities
                 logits = self.modules.ctc_lin(x)
                 p_ctc = self.hparams.log_softmax(logits)
-                return p_ctc, p_seq, wav_lens
-            else:
-                return p_seq, wav_lens
         else:
-            p_tokens, scores = self.hparams.beam_searcher(x, wav_lens)
-            return p_seq, wav_lens, p_tokens
+            topk_tokens, topk_lens, _, _ = self.hparams.beam_searcher(
+                x, wav_lens
+            )
+
+            # Select the best hypothesis
+            best_hyps, best_lens = topk_tokens[:, 0, :], topk_lens[:, 0]
```

Comment:
Why do we keep returning the best hypotheses like this as a standard? It makes the code harder to read for an edge use case.

Reply:
fixed